lightning/src/ln/channelmanager.rs

```diff
@@ -489,19 +507,38 @@ impl PendingOutboundPayment {
 		core::mem::swap(&mut session_privs, match self {
 			PendingOutboundPayment::Legacy { session_privs } |
 			PendingOutboundPayment::Retryable { session_privs, .. } |
-			PendingOutboundPayment::Fulfilled { session_privs, .. }
-				=> session_privs
+			PendingOutboundPayment::Fulfilled { session_privs, .. } |
+			PendingOutboundPayment::RetriesExceeded { session_privs, .. }
+				=> session_privs,
 		});
 		let payment_hash = self.payment_hash();
 		*self = PendingOutboundPayment::Fulfilled { session_privs, payment_hash };
 	}
 
+	fn mark_retries_exceeded(&mut self) -> Result<(), ()> {
+		let mut session_privs = HashSet::new();
+		let our_payment_hash;
+		core::mem::swap(&mut session_privs, match self {
+			PendingOutboundPayment::Legacy { .. } |
+			PendingOutboundPayment::Fulfilled { .. } =>
+				return Err(()),
+			PendingOutboundPayment::Retryable { session_privs, payment_hash, .. } |
+			PendingOutboundPayment::RetriesExceeded { session_privs, payment_hash, .. } => {
+				our_payment_hash = *payment_hash;
```

Comment:
Consider setting outside using `self.payment_hash()` like in `mark_fulfilled` to avoid a side-effect in the match.

Reply:
I didn't bother cause it implies unwrapping the payment_hash when we don't need to. If you feel strongly I can change it.

Reply:
Ah, didn't notice that difference. Don't feel too strongly. Just felt a little inconsistent when I saw it.